### PR TITLE
8292911: Add a note about jtreg failure handlers in the testing doc

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this change to the testing documentation which adds a note about failure handlers in jtreg? This addresses https://bugs.openjdk.org/browse/JDK-8292911.

Recently, there have been users who have been prompted for sudo password when running jtreg tests locally and those tests running into failures. This is because one of the commands that gets executed by the jtreg failure handler uses sudo (the `sudo dmesg` command on macos).

This commit updates the `testing.md` to explain what the failure handler does and also mentions the configuration that can be used to disable the failure handler. This commit doesn't specify how to configure the `sudoers` file if the user still wants to allow the failure handlers to run, because I thought that would be outside the scope of this documentation.

I attempted to auto generate the `testing.html` from the changes to `testing.md` using the `make update-build-docs` command, but that generated a lot of unnecessary diffs not just in `testing.html` but other html files in the `doc` directory. So, as suggested in a different PR which involved updates to these files, I manually added the section to the `testing.html` too.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292911](https://bugs.openjdk.org/browse/JDK-8292911): Add a note about jtreg failure handlers in the testing doc

### Reviewers
 * [Lance Andersen](https://openjdk.org/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Jonathan Gibbons](https://openjdk.org/census#jjg) (@jonathan-gibbons - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10021/head:pull/10021` \
`$ git checkout pull/10021`

Update a local copy of the PR: \
`$ git checkout pull/10021` \
`$ git pull https://git.openjdk.org/jdk pull/10021/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10021`

View PR using the GUI difftool: \
`$ git pr show -t 10021`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10021.diff">https://git.openjdk.org/jdk/pull/10021.diff</a>

</details>

### CODE REVIEW

File: doc/testing.md [hunk 1]
```diff
@@ -43,6 +43,17 @@ containing the core JMH and transitive dependencies. The recommended
...
 after which `--with-jmh=build/jmh/jars` should work.
 
+When tests fail or timeout, jtreg runs its failure handler to capture necessary
+data from the system where the test was run. This data can then be used to
+analyze the test failures. Collecting this data involves running various commands
+(which are listed in files residing in `test/failure_handler/src/share/conf`)
+and some of these commands use `sudo`. If the system's `sudoers` file isn't
+configured to allow running these commands, then it can result in password being
+prompted during the failure handler execution. Typically, when running locally,
+collecting this additional data isn't always necessary. To be able to disable
```
Comment (erikj79) [inline doc/testing.md:53]: I would suggest removing `be able to` from the last sentence, I think it reads better. Otherwise good.

**Comment by erikj79** [comment]
Inside Oracle, if you configure using Jib you should get the same version of pandoc that was used to generate the existing html files.

File: doc/testing.md (hunk 1 above)
Comment (jaikiran) [reply doc/testing.md]: Fixed in the updated version of this PR.

**Comment by jaikiran** [comment]
Thank you Erik for your review. I've updated the PR to include your suggested change. Additionally, I received another suggestion, offline, that it would be good to include a minimal example of configuring sudoers file if the user still wants to let the failure handler to run, without being prompted for password. I've updated the text to include that detail.

**Comment by jaikiran** [comment]
> Inside Oracle, if you configure using Jib you should get the same version of pandoc that was used to generate the existing html files.

I gave that a try (and verified that it uses the jib installed pandoc), but it still ends up generating way too many changes on existing untouched content. I did however, let it generate that content and then manually copied over only the lines of interest in this PR.

**Comment by erikj79** [comment]
> > Inside Oracle, if you configure using Jib you should get the same version of pandoc that was used to generate the existing html files.
> 
> I gave that a try (and verified that it uses the jib installed pandoc), but it still ends up generating way too many changes on existing untouched content. I did however, let it generate that content and then manually copied over only the lines of interest in this PR.

That's strange. I have just verified this on both Macos and Linux, and I get zero spurious changes from `make update-build-docs`.

File: doc/testing.md [hunk 2]
```diff
@@ -43,6 +43,29 @@ containing the core JMH and transitive dependencies. The recommended
...
+When tests fail or timeout, jtreg runs its failure handler to capture necessary
+data from the system where the test was run. This data can then be used to
+analyze the test failures. Collecting this data involves running various commands
+(which are listed in files residing in `test/failure_handler/src/share/conf`)
+and some of these commands use `sudo`. If the system's `sudoers` file isn't
+configured to allow running these commands, then it can result in password being
+prompted during the failure handler execution. Typically, when running locally,
+collecting this additional data isn't always necessary. To disable running the
+failure handler, use `--enable-jtreg-failure-handler=no` when running `configure`.
+If, however, you want to let the failure handler to run and don't want to be prompted
```
Comment (erikj79) [inline doc/testing.md:55]: Most of this file is kept close to 80 chars per line, at least in the text blocks. Up until here, you have kept with this style, but the additions below get increasingly wider, which looks a bit weird. Would you mind tiding it up a bit?

Comment (jaikiran) [reply doc/testing.md]: Hello Erik, I've now updated the file to fix the line lengths.

**Comment by jaikiran** [comment]
> > > Inside Oracle, if you configure using Jib you should get the same version of pandoc that was used to generate the existing html files.
> > 
> > 
> > I gave that a try (and verified that it uses the jib installed pandoc), but it still ends up generating way too many changes on existing untouched content. I did however, let it generate that content and then manually copied over only the lines of interest in this PR.
> 
> That's strange. I have just verified this on both Macos and Linux, and I get zero spurious changes from `make update-build-docs`.

I'll send the diff that gets generated for me and other details, privately to you, to see if can understand what's going on.

**Review by LanceAndersen — APPROVED** [review]

**Review by erikj79 — APPROVED** [review]

**Review by jonathan-gibbons — APPROVED** [review]
The general nature of the text seems OK; it's good to give a reason why you may not want/need the failure handler to run, as well suggesting how to make it work when you do want to run the failure handler.

**Comment by jaikiran** [comment]
Thank you everyone for the reviews.

**Pull request closed without merging**